button-merchant/src/main/java/com/usebutton/merchant/ConnectionManagerImpl.java

```diff
@@ -90,6 +91,9 @@ static ConnectionManager getInstance(String baseUrl, String userAgent,
     @Override
     public void setApplicationId(String applicationId) {
         this.applicationId = applicationId;
+
+        // Update the endpoint to the App ID flavor
+        baseUrl = String.format(ButtonMerchant.FMT_BASE_URL_APP_ID, applicationId);
```

Comment:
I suppose you could also check it here and not use it if it's invalid.

Reply:
No need, it'll only ever come through `ButtonMerchant.configure(~)` -> `ButtonInternalImpl.configure(~)`